CMakeLists.txt

```diff
@@ -1,90 +1,65 @@
-option(CREATE_DOC "Whether or not to create doxygen doc target.")
-
-# Offer the user the choice of overriding the installation directories
-set(INSTALL_LIB_DIR lib CACHE PATH "Installation directory for libraries")
-set(INSTALL_BIN_DIR bin CACHE PATH "Installation directory for executables")
-set(INSTALL_INCLUDE_DIR include CACHE PATH
-  "Installation directory for header files")
+cmake_minimum_required(VERSION 3.3)
+project(fccedm VERSION 0.5.2)
+set(fccedm_VERSION "${fccedm_VERSION_MAJOR}.${fccedm_VERSION_MINOR}")
+
+# Define basic build settings
+# Use GNU-style hierarchy for installing build products
+include(GNUInstallDirs)
+
+# Define a default build type can be overriden by passing
+# ``-DCMAKE_BUILD_TYPE=<type>`` when invoking CMake
+if(NOT CMAKE_CONFIGURATION_TYPES)
+  if(NOT CMAKE_BUILD_TYPE)
+    set(CMAKE_BUILD_TYPE RelWithDebInfo
+      CACHE STRING "Choose the type of build, options are: None Release MinSizeRel Debug RelWithDebInfo"
+      FORCE
+      )
+  else()
+    set(CMAKE_BUILD_TYPE "${CMAKE_BUILD_TYPE}"
+      CACHE STRING "Choose the type of build, options are: None Release MinSizeRel Debug RelWithDebInfo"
+      FORCE
+      )
+  endif()
+endif()
 
+# Require c++14 standard and flags
+set(CMAKE_CXX_STANDARD 14)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+set(CMAKE_CXX_EXTENSIONS OFF)
+set(CMAKE_CXX_FLAGS "-fPIC -DDROP_CGAL -Wall -Wextra -Wpedantic -Wno-unused-variable -Wno-unused-parameter")
 
+# Use .so also on macOS so ROOT can load them
 if(APPLE)
   set(CMAKE_SHARED_LIBRARY_SUFFIX ".so")
-endif(APPLE)
-
-set(DEF_INSTALL_CMAKE_DIR cmake)
-set(INSTALL_CMAKE_DIR ${DEF_INSTALL_CMAKE_DIR} CACHE PATH
-  "Installation directory for CMake files")
-
-# Make relative paths absolute (needed later on)
-foreach(p LIB BIN INCLUDE CMAKE)
-  set(var INSTALL_${p}_DIR)
-  if(NOT IS_ABSOLUTE "${${var}}")
-    set(${var} "${CMAKE_INSTALL_PREFIX}/${${var}}")
-  endif()
-endforeach()
-
-# Set up C++14
-set(CPP_STANDARD_FLAGS "-std=c++14")
-if (${APPLE})
-    set(CPP_STANDARD_FLAGS "-std=c++14\ -stdlib=libc++")
 endif()
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC -DDROP_CGAL ${CPP_STANDARD_FLAGS} -Wall -Wextra -Wpedantic -Wno-unused-variable -Wno-unused-parameter")
 
 
-# Make sure we find the Find*.cmake functions distributed with this package
-set(CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
-find_package(podio REQUIRED HINTS $ENV{PODIO})
-message(STATUS "Found podio: ${podio_DIR}")
-# Make sure the library is found.
-# Not the case if LD_LIBRARY_PATH is wrong
-find_library(PODIOLIB podio PATHS $ENV{PODIO}/lib)
-if (NOT PODIOLIB)
-    message(FATAL_ERROR "libpodio.so(dylib) cannot be found dynamically. Make sure you have sourced PODIO init*.sh file to set up your environment to use PODIO")
-endif()
-link_directories(${podio_LIBRARY_DIR})
+# Regenreate datamodel when invoking cmake
+execute_process(COMMAND python $ENV{PODIO}/python/podio_class_generator.py edm.yaml datamodel datamodel
```

Comment:
Technically, it is before any `find_package` call, but I get your point. Moved it down and use the podio variable instead. For the record, it would be nice to have [this](https://github.com/hegner/podio/pull/29/files#diff-533f28e8a4609b6acedba0da0d47372bR23) to make it even nicer :)